Sources/Core/Services/NowPlayingService.swift

```diff
@@ -45,12 +46,20 @@ final class ModernAVPlayerNowPlayingService: NowPlaying {
     private var task: URLSessionTask?
 
     func update(metadata: PlayerMediaMetadata?, duration: Double?, isLive: Bool) {
-        infos = parseInfos(metadata: metadata, duration: duration)
+        infos = parse(metadata: metadata)
         if #available(iOS 10, *) {
             infos[MPNowPlayingInfoPropertyIsLiveStream] = isLive
         }
+        if let duration = duration,
+            duration.isNormal {
+            infos[MPMediaItemPropertyPlaybackDuration] = duration
+        }
         MPNowPlayingInfoCenter.default().nowPlayingInfo = infos
     }
+
+    func update(metadata: PlayerMediaMetadata) {
+        MPNowPlayingInfoCenter.default().nowPlayingInfo = parse(metadata: metadata)
```

Comment:
instance infos variable should be also set with new metadata.
we have to think again about the parse method

Reply:
What about that new commit I just push?